contractcourt/breach_arbitrator.go

```diff
@@ -1539,19 +1596,43 @@ func (b *BreachArbitrator) sweepSpendableOutputsTxn(txWeight lntypes.WeightUnit,
 	}
 	txFee := feePerKw.FeeForWeight(txWeight)
 
+	// At this point, we'll check to see if we have any extra outputs to
+	// add from the aux sweeper.
+	extraChangeOut := fn.MapOptionZ(
+		b.cfg.AuxSweeper,
+		func(aux sweep.AuxSweeper) fn.Result[sweep.SweepOutput] {
+			return aux.DeriveSweepAddr(inputs, pkScript)
+		},
+	)
+	if err := extraChangeOut.Err(); err != nil {
+		return nil, err
+	}
+
 	// TODO(roasbeef): already start to siphon their funds into fees
 	sweepAmt := int64(totalAmt - txFee)
 
 	// With the fee calculated, we can now create the transaction using the
 	// information gathered above and the provided retribution information.
 	txn := wire.NewMsgTx(2)
 
-	// We begin by adding the output to which our funds will be deposited.
+	// First, we'll add the extra sweep output if it exists, subtracting the
+	// amount from the sweep amt.
+	if b.cfg.AuxSweeper.IsSome() {
+		extraChangeOut.WhenResult(func(o sweep.SweepOutput) {
+			sweepAmt -= o.Value
```

Comment:
is it possible that `o.Value` is greater than or equal to `sweepAmt`?

Reply:
imo we should be defensive here in case the call to `DeriveSweepAddr` returns something funky

Reply:
So with the way things work as is, the current o.Value will always be a static amount that's above dust, and we won't try to sweep anything that is around that dust level. 

With that said, I think it makes sense to add a more defensive check here.